### ISSUE TITLE
`no-extra-parens`: false positive on deprecated option `allowNodesInSpreadElement`

### ISSUE THREAD

**Issue by burtek** [eslint-stylistic/eslint-stylistic#979]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Config:

```
'@stylistic/no-extra-parens': ['error', 'all', {}]
```

Error in eslint CLI output:
> 2025-09-09 15:23:15.853 [error] [@stylistic/eslint-plugin]: You are using deprecated option("allowNodesInSpreadElement") in "no-extra-parens", please use "ignoreNodes" instead.

Cause:
https://github.com/eslint-stylistic/eslint-stylistic/blob/8ccd98090d8a19ce9d96044693f28ca05605508a/packages/eslint-plugin/rules/no-extra-parens/no-extra-parens.ts#L146-L148
https://github.com/eslint-stylistic/eslint-stylistic/blob/8ccd98090d8a19ce9d96044693f28ca05605508a/packages/eslint-plugin/rules/no-extra-parens/no-extra-parens.ts#L154-L155

`Set` is always truthy even if empty.

### Reproduction

https://stackblitz.com/edit/github-xtfmuv8f?file=eslint.config.js

<img width="881" height="128" alt="Image" src="https://github.com/user-attachments/assets/f6fbfdb8-dd38-43c3-aae2-33f1639add49" />

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

I might do those at some point later on, unless someone else does it earlier, but I don't wanna block someone else if they get to it first, especially since I don't know when/if I'm gonna have time for it

**Comment by 9romise** [comment]
Thanks for bringing this up! Will fix in the next major. See #967
